### DISCUSSION TITLE
AI-deck 1.1 run helloworld example failed with JTAG (Olimex ARM-USB-TINY-H)

### DISCUSSION THREAD

**Discussion by c2wong** [bitcraze/discussions#1705] [Q&A] [answered]
Hi there, 
        I've been working with AI-deck 1.1, combined with Olimex ARM-USB-TINY-H + ARM-JTAG-20-10 converter as suggested.  I follow the instruction to attach them like this:
[<blockquote class="imgur-embed-pub" lang="en" data-id="kiuH999"><a href="https://imgur.com/kiuH999">View post on imgur.com</a></blockquote>](url)

I try to use the docker file and also build my own platfom, but both stuck on build and run hello world example on board.  The operation system is Ubuntu 18.04.6 LTS VMware running on Windows 10.

**The JTAG device pop up correctly in my Ubuntu 18.04 virtual machine with lsusb:**
```
$ lsusb
Bus 001 Device 006: ID 15ba:002a Olimex Ltd. ARM-USB-TINY-H JTAG interface
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 002 Device 003: ID 0e0f:0002 VMware, Inc. Virtual USB Hub
Bus 002 Device 002: ID 0e0f:0003 VMware, Inc. Virtual Mouse
Bus 002 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
```

**The docker images:**
```
$ docker image ls
REPOSITORY              TAG       IMAGE ID       CREATED         SIZE
aideck-with-autotiler   latest    415ff5d76f37   6 minutes ago   9.41GB
bitcraze/aideck         latest    9654f385dd60   6 months ago    9.29GB
bitcraze/aideck         3.8.1     986d55c4ba3b   3 years ago     4.76GB
```

**The command I use:**
```
$ docker run --rm -v ${PWD}:/module --device /dev/ttyUSB0 --privileged -P bitcraze/aideck tools/build/make-example examples/other/hello_world_gap8 run
```
(I change  the sample command from flash to run)

**The error message:**
```
$ docker run --rm -v ${PWD}:/module --device /dev/ttyUSB0 --privileged -P aideck-with-autotiler tools/build/make-example examples/other/hello_world_gap8 run
Collecting numpy==1.22.3
  Using cached numpy-1.22.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.8 MB)
ERROR: imageio 2.17.0 has requirement pillow>=8.3.2, but you'll have pillow 7.2.0 which is incompatible.
Installing collected packages: numpy
  Attempting uninstall: numpy
    Found existing installation: numpy 1.24.3
    Uninstalling numpy-1.24.3:
      Successfully uninstalled numpy-1.24.3
Successfully installed numpy-1.22.3
gapy --target=ai_deck --platform=board --work-dir=/module/examples/other/hello_world_gap8/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --exec-prepare --exec --binary=/module/examples/other/hello_world_gap8/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/hello_world_gap8
Open On-Chip Debugger 0.10.0+dev-gb84d97ec4 (2022-04-20-14:26)
Licensed under GNU GPL v2
For bug reports, read
        http://openocd.org/doc/doxygen/bugs.html
debug_level: 0

DEPRECATED! use 'adapter speed' not 'adapter_khz'
DEPRECATED! use 'adapter driver' not 'interface'
TARGET create
0
GAP8 INIT TARGET
Error: JTAG scan chain interrogation failed: all ones
Error: Check JTAG interface, timings, target power, etc.
Error: Trying to use configured scan chain anyway...
Error: gap8.cpu: IR capture error; saw 0x0f not 0x01
GAP8 examine target
Init jtag
Initialising GAP8 JTAG TAP
Error: Burst read failed
Error: Burst read failed
^Ccan't read halt state
can't read halt state
Init jtag
Initialising GAP8 JTAG TAP
```

Can you tell me what's wrong?  Is there specific version criteria or modifications should be made?  How to supply power to standalone AI-deck 1.1 module?  Do you have AI-deck 1.1 schematics we can reference?

Thanks a lot.

CC Wong

**Comment by gemenerik** [comment, marked as answer]
Hi CC Wong,

Thanks for your detailed post—you're making things much easier! It seems like your cable connection is correct. While it probably won’t affect the errors you’re seeing, have you tried using the `flash` command instead of `run`? 

This is an error you would expect to see if the debugger cannot talk to the GAP8. Could you confirm that the AI-deck is properly attached to the Crazyflie? And that the Crazyflie is powered by a charged battery, and it’s turned on? The debugger doesn’t power the chip.

//Rik

**Reply by c2wong** [reply]
Hi Rik,

Thank you for instant reply.  I think you've answered my question.  We didn't get the Crazyflie, because we bought AI-deck for image collection purposes only.  I will try to look for schematics again on the product page.  Has your customer tried to supply AI-deck power from different source other than CrazyFlie?

Just in case, it seems there is no Crazyflie in stock from your online store information, how much lead time before we can buy one?

Appreciate for your time.

Best Regards,
CC Wong

**Reply by ToveRumar** [reply]
Hi @c2wong!
Im jumping in to the thread here. We do have Crazyflie 2.1+  in stock, so please feel free to put in an order for anything you need. 
If you have any further questions regarding ordering or stock levels, please contact us at contact@bitcraze.io.
